dragonfly/block/glass_pane.go

```diff
@@ -0,0 +1,66 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/dragonfly/entity/physics"
+	"github.com/df-mc/dragonfly/dragonfly/item/tool"
+	"github.com/df-mc/dragonfly/dragonfly/world"
+	"github.com/go-gl/mathgl/mgl64"
+)
+
+// GlassPane is a transparent block that can be used as a more efficient alternative to glass blocks.
+type GlassPane struct{}
+
+// BreakInfo ...
+func (p GlassPane) BreakInfo() BreakInfo {
+	return BreakInfo{
+		Hardness: 0.3,
+		Harvestable: func(t tool.Tool) bool {
+			return true // TODO(lhochbaum): Glass panes can be silk touched, implement silk touch.
+		},
+		Effective: nothingEffective,
+		Drops:     simpleDrops(),
+	}
+}
+
+// EncodeItem ...
+func (p GlassPane) EncodeItem() (id int32, meta int16) {
+	return 102, meta
+}
+
+// EncodeBlock ...
+func (p GlassPane) EncodeBlock() (name string, properties map[string]interface{}) {
+	return "minecraft:glass_pane", map[string]interface{}{}
+}
+
+// AABB adjusts bounding box of the glass pane.
+func (p GlassPane) AABB(pos world.BlockPos, w *world.World) []physics.AABB {
+	return calculateThinBounds(pos, w)
+}
+
+// ConnectsWith ...
+func (p GlassPane) ConnectsWith(other world.Block) bool {
+	switch other.(type) {
+	case GlassPane, StainedGlassPane: // TODO(lhochbaum): Add iron bars as soon as they're implemented.
```

Comment:
Note that glass panes also connect with most other blocks. We probably need some interface to determine if a block has a specific solid face, where not implementing it by default means it is solid on all sides.